Commit failure does not trigger error

sek3b opened the pull request:
The function router.commit() does not produce an err when routing engine fails configuration. From what i have seen on an mx480 result.rpc_reply.commit_results.routing_engine.rpc_error will be present when one of the routing engine fails and if it passes it does not exist.

shayneeking committed:
Commit failure does not trigger error

The function router.commit() does not produce an err when routing engine fails configuration. From what i have seen on an mx480 result.rpc_reply.commit_results.routing_engine.rpc_error will be present when one of the routing engine fails and if it passes it does not exist.

shayneeking committed:
Creating a Private Copy of the Configuration

Per http://www.juniper.net/techpubs/en_US/junos/information-products/pathway-pages/junos-xml-management-protocol/junos-xml-management-protocol.pdf
Created functions openPrivate and closePrivate one which opens the config in private mode (configure private) and one which closes the config similar to a rollback but "Any changes to the private copy are lost".

shayneeking committed:
Update netconf.js

shayneeking committed:
Update netconf.js

darylturner commented:
Hi - sorry took me a while to answer this, been away on paternity leave. Would it make more sense to check this in netconf.js and wrap the clients callback with error checking? This is how i do it in the load method for example.

shayneeking committed:
Cisco implementation

Added head back which is required for cisco.
Added comp for urn:ietf:params:netconf:base:1.0
Modified junosCommit to conform to daryl suggestion (https://github.com/darylturner/node-netconf/pull/3)
Converted commands to OS specific commands

darylturner commented:
These changes would have to be published to NPM under a new major revision as breaks semver.
